bsc.typechecker/higherrank/higherrank.exp

```diff
@@ -5,5 +5,12 @@ test_veri_only HigherRank
 test_veri_only Undefined
 test_veri_only DeepSeqCond
 
-# Use of an ininitialized value results in an expected error
+# Use of a higher-rank member of an unininitialized struct value
+# results in an expected error
 compile_verilog_fail_error Uninitialized.bs G0028 1 sysUninitialized
+compare_file Uninitialized.bs.bsc-vcomp-out
+
+# Use of a member of a higher-rank member of an uninitialized struct value
```

Comment:
Is this comment still accurate?  I'm not sure that I see anything in the output that specifically says anything about which struct field is uninitialized.

Also, what is being tested with these two tests?  These are worth having, but I think I'd prefer tests written in BSV that start with an uninitialized variable and then attempt to define the structure by parts, and we check that BSC does __not__ report a use of an uninitialized value.  That is, something like this:
```
Foo f;
f.x.fst = ?;
```
This desugars into `new_f = old_f { x: old_f.x { fst: ? } }` which references `f.x`.  If the instance for `Foo` is a struct with a raw uninitialized value for `x`, then BSC will warn about a use of an uninitialized value (which we don't want); if BSC instead further expands the structure of `x` to assign it a pair with uninitialized fields, then we won't get a warning (which is what we want).  By looking for a warning, we can test how PrimDeepSeqCond instances are created; I don't think the current tests are testing that.  (It might also be worthwhile to have BSV tests that do make use of uninitialized fields, to make sure that the warning messages look OK.  For example, I notice that the position is "Unknown" in these Classic/BH examples; hopefully the position would be better in BSV examples?)

Reply:
Ah yes, I fixed the implementation to not use a raw uninitialized value but didn't update the comment.  

Also added a test as you suggested.  

As mentioned in my above reply, the last to arguments on `compile_verilog_fail_error` do in fact appear to be needed.